### PR TITLE
Convert polyfill to use gl-matrix directly

### PR DESCRIPTION
**Pull request by toji**
Verified that this ends up having effectively zero impact on size due to tree shaking. Using the library directly rather than porting individual functions over piecemeal will make it easier to add more mathematically challenging features in the future, such as arm modeling for 3DoF controllers (which I have almost completed porting from Boris' RayInput library.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "eslint": "^4.16.0",
     "eslint-config-google": "^0.9.1",
     "fast-async": "^6.3.0",
+    "gl-matrix": "^2.5.1",
```
Comment (jsantell) [inline package.json]: Should be a `dependency`, not a `devDependency`

File: src/api/XRDevicePose.js [hunk 2]
```diff
@@ -14,7 +14,7 @@
  */
 
 const PRIVATE = Symbol('@@webxr-polyfill/XRDevicePose');
-import { mat4_identity } from '../math';
+import * as mat4 from '../../node_modules/gl-matrix/src/gl-matrix/mat4';
```
Comment (jsantell) [inline src/api/XRDevicePose.js]: Does `import * as mat4 from 'gl-matrix/src/gl-matrix/mat4'` result in the same treeshaked result? That seems preferable to the relative node_modules path

File: package.json (hunk 1 above)
Comment (jsantell) [inline package.json]: Should be under `dependencies` instead of `devDependencies`

File: src/api/XRDevicePose.js (hunk 2 above)
Comment (jsantell) [inline src/api/XRDevicePose.js]: Does `gl-matrix/src/gl-matrix/mat4` work? I think that's preferable to the relative node_modules path

**Comment by jsantell** [comment]
Oops, double review comments above -- I attempted this before but couldn't remember what issues I ran into -- checking the tests there's a failure at:  `(function (exports, require, module, __filename, __dirname) { import * as glMatrix from "./common.js";`

Looks like we'll need to manually transform the dependency unfortunately -- I hope that not excluding gl-matrix [here](https://github.com/immersive-web/webxr-polyfill/blob/14315c4910dc66a85b80bd8a9f44654baf00ac28/rollup.config.js#L38) -- and suspect the module build may have an issue as well. If this works I'll be very exciting :)

**Comment by jsantell** [comment]
With #18's build changes, this should now work

**Comment by toji** [comment]
Rebased this patch onto #18 and fixed the tests. Thank for updating the build system!

**Comment by jsantell** [comment]
I think the commits need a squash (due to different ordering from master), but other than that, should be good to land!

**Comment by toji** [comment]
Squashed and merged!

**Pull request merged**